src/main/java/com/thealgorithms/scheduling/SpeculativeExecutionScheduling.java

```diff
@@ -0,0 +1,82 @@
+package com.thealgorithms.scheduling;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * SpeculativeExecutionScheduling runs multiple copies of the same task in parallel,
+ * picking the first one that finishes successfully. It is used to mitigate the
+ * impact of stragglers (slow tasks).
+ *
+ * Use Case: Big data systems like Hadoop and Spark, where it helps improve job completion time.
+ *
+ * @author Hardvan
+ */
+public final class SpeculativeExecutionScheduling {
+
+    static class Task {
+        String name;
+        boolean completed;
+        long startTime;
+
+        Task(String name) {
+            this.name = name;
+            this.completed = false;
+            this.startTime = -1;
```

Comment:
It should be the time when the task arrives to the system and becomes available for execution.